src/plugins/custom_integrations/server/types.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CustomIntegration, CategoryCount } from '../common';
+
+export interface CustomIntegrationsPluginSetup {
+  registerCustomIntegration(customIntegration: CustomIntegration): void;
+  getAddableCustomIntegrations(): CustomIntegration[];
+  getAddableCategories(): CategoryCount[];
+  getReplaceableCustomIntegrations(): CustomIntegration[];
```

Comment:
Can we add some comments that explain the difference between 'addable' and 'replaceable'? I also wonder if `append` and `replacement` would make slightly more sense.